### PR TITLE
Dive bombs

### PR DESCRIPTION
**Pull request by gknisely**
# Issue

Updated default speeds for urban areas based on roadclass for the enhancer.  Also, updated default speeds based on roadclass in lua.  

Fixed an issue where we were subtracting 1 from uint32_t when 0 for stop impact.  

Updated reclassify link logic to allow residential roads to be added to the tree, but we only downgrade the links to tertiary.  

Updated TransitionCost functions to add 1.5 to the turncost when transitioning from a ramp to a non ramp and vice versa. Also, added 0.5f to the turncost if the edge is a roundabout.

Paired with @dgearhart on logic.  

## Tasklist

 - [ ] Review - you must request approval to merge any PR to master
 - [x] Generally use squash merge to rebase and clean comments before merging
 - [x] Update the [changelog](CHANGELOG.md)

## Requirements / Relations

NA

### Before: Major interchange
image

### After: Major interchange
image

### Before: Major interchange # 2
image

### After: Major interchange # 2
image

### Before: Rural interchange
image

### After: Rural interchange
image

### Before: Rural interchange # 2
image

### After: Rural interchange # 2
image

### Before: Tagged speed interchange
image

### After: Tagged speed interchange
image

### CODE REVIEW

File: src/mjolnir/graphenhancer.cc [hunk 1]
```diff
@@ -68,6 +68,7 @@ constexpr float kDensityLatDeg = (kDensityRadius * kMetersPerKm) / kMetersPerDeg
 constexpr float kTurnChannelFactor = 1.25f;
 constexpr float kRampDensityFactor = 0.8f;
 constexpr float kRampFactor = 0.85f;
```
Comment (dgearhart) [inline src/mjolnir/graphenhancer.cc]: we discussed setting kRampFactor to 0.7. I don't think you need kRampDensityFactor

File: src/mjolnir/pbfgraphparser.cc [hunk 2]
```diff
@@ -1366,7 +1366,8 @@ struct graph_callback : public OSMPBF::Callback {
           from_way_id = member.member_id;
         } else if (member.role == "to" &&
                    member.member_type == OSMPBF::Relation::MemberType::Relation_MemberType_WAY) {
-          restriction.set_to(member.member_id);
+          if (!restriction.to())
```
Comment (gknisely) [inline src/mjolnir/pbfgraphparser.cc]: This was a fix for when old editors allow people to enter 2 or more to roles for the end of a restriction.  We just take the first one and toss the rest.

File: src/mjolnir/linkclassification.cc [hunk 3]
```diff
@@ -227,7 +227,11 @@ std::pair<uint32_t, uint32_t> Reclassify(LinkTreeNode& root,
       sequence<Edge>::iterator element = edges[idx];
       auto edge = *element;
       if (rc > edge.attributes.importance) {
-        edge.attributes.importance = rc;
+        if (rc < static_cast<uint32_t>(RoadClass::kUnclassified))
+          edge.attributes.importance = rc;
+        else
+          edge.attributes.importance = static_cast<uint32_t>(RoadClass::kTertiary);
```
Comment (kevinkreiser) [inline src/mjolnir/linkclassification.cc]: whats this doing?

File: src/mjolnir/linkclassification.cc [hunk 4]
```diff
@@ -295,7 +298,7 @@ void ReclassifyLinks(const std::string& ways_file,
     // Do not continue if this link intersects a "major" road
     // TODO - might need to continue to expand...but will need to determine
     // if the link enters back onto the same road that was exited.
-    if (bundle.node.non_link_edge_ && rc < static_cast<uint32_t>(RoadClass::kUnclassified)) {
+    if (bundle.node.non_link_edge_ && rc <= static_cast<uint32_t>(RoadClass::kResidential)) {
```
Comment (kevinkreiser) [inline src/mjolnir/linkclassification.cc]: this change is semantically equivallent, i suppose it easier to read?

File: src/mjolnir/linkclassification.cc (hunk 3 above)
Comment (gknisely) [reply src/mjolnir/linkclassification.cc]: comments were in the description.."Updated reclassify link logic to allow residential roads to be added to the tree, but we only downgrade the links to tertiary."

**Pull request merged**